### ISSUE TITLE
In-browser work?

### ISSUE THREAD

**Issue by blokhin** [codeforscience/sciencefair#182]
Hi, I see it's based on Electron, which means at least theoretically it can have an in-browser GUI. Is that true?

**Comment by bencevans** [comment]
There's potential. Sciencefair relies on Dat which I don't think can run directly in the browser currently however there are developments with proxies/gateways to do so.

**Comment by karissa** [comment]
Dat works in the browser with webrtc and Gateways, see dat-js

**Comment by blahah** [comment]
@blokhin indeed, part of the strategy for sciencefair has always been to ensure that the experience users have in the app can (eventually) easily be integrated into any platform on any device that supports the web. Another part of the strategy is to ensure that the technologies we depend on are as stable and straightforward as possible. These are not just goals but are actively featured into every major decision about the project. For example:

- we use choo as the app framework because it's extremely lightweight and stable, requires no special tools and runs wherever modern JS works
- we use nanocomponents because they run everywhere, integrate with everything, and are understandable to anyone who has learned basic web development
- we use browserify because it is vastly more straightforward and stable than any alternative 'bundling' technologies
- we actively avoid React, Webpack, and anything else that is hyper-trendy and endlessly evolving

Currently sciencefair is not in active development, but when development does re-start (and it will), we will need to work on separating out non-browser APIs from the client-side code before any significant part of it can really be re-used in the browser. There are a bunch of things that depend on Electron's renderer-process node integration, but shouldn't, that we need to refactor.
